src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,124 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_CAPACITY = 10;
+    private static final int DEFAULT_SIZE = 0;
+    private int quantity = DEFAULT_SIZE;
+    private Object[] elements = new Object[DEFAULT_CAPACITY];
+
     @Override
     public void add(T value) {
-
+        growIfArrayFull();
+        elements[quantity] = value;
+        quantity += 1;
     }
 
     @Override
     public void add(T value, int index) {
-
+        checkInRangeForAddition(index);
+        growIfArrayFull();
+        Object [] temp = new Object[elements.length];
+        System.arraycopy(elements, 0, temp, 0, temp.length);
+        for (int i = 0, j = 0; i <= quantity; i++, j++) {
+            if (i == index) {
+                elements[i] = value;
+                i++;
+            }
+            elements[i] = temp[j];
+        }
+        quantity += 1;
     }
 
     @Override
     public void addAll(List<T> list) {
-
+        for (int i = 0; i < list.size(); i++) {
+            add(list.get(i));
+        }
     }
 
     @Override
     public T get(int index) {
-        return null;
+        checkInRangeForGet(index);
+        return (T) (elements[index]);
     }
 
     @Override
     public void set(T value, int index) {
-
+        checkInRangeForGet(index);
+        elements[index] = value;
     }
 
     @Override
     public T remove(int index) {
-        return null;
+        checkInRangeForGet(index);
+        Object [] temp = new Object[elements.length];
+        System.arraycopy(elements, 0, temp, 0, elements.length);
```

Comment:
Again, this is a redundant variable and unnecessary array copying. You can simply shift all elements to the left of the index you want to remove. This will make your code more efficient.